subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,466 @@
+                    page_link = link_tag['href']
+                    subtitle_id = (lambda m: m.group(1) if m else None)(subtitle_href_re.search(page_link))
+                    multiple_episodes = bool(div.find_parent('div', {'class': 'pack'}) or
+                                             pack_name_re.findall(link_tag.string))
+                    featured = bool(div.find_parent('div', {'class': 'destaque'}))
+                    no_downloads = (lambda v: int(v) if v and v.isdigit() else None)(
+                        (lambda m: m.group(1) if m else None)(rating_info_re.search(div.text)))
+                    rating = (lambda v: int(v) if v and v.isdigit() else None)(
+                        (lambda m: m.group(2) if m else None)(rating_info_re.search(div.text)))
+                    timestamp = (lambda d: datetime.strptime(d, '%d/%m/%Y - %H:%M') if d else None)(
+                        (lambda m: m.group(1) if m else None)(timestamp_info_re.search(div.text)))
+
+                    # Using the candidate name to filter out bad candidates
+                    # (wrong type, wrong episode, wrong season or even wrong title)
+                    guess = guess_file_info(candidate_name + '.mkv', type=title.get('type'))
+                    if not self.matches(params, guess, ignore_episode=multiple_episodes):
+                        continue
+
+                    # Unfortunately, the only possible way to know the release names of a specific candidate is to
+                    # download the compressed file (rar/zip) and list the file names.
+                    subtitle_names = self.get_subtitle_names(subtitle_id, timestamp)
+
+                    if not subtitle_names:
+                        continue
+
+                    for name in subtitle_names:
+                        # Filtering out bad candidates (one archive might contain subtitles for the whole season,
+                        # therefore this filtering is necessary) and some subtitles are in a relative folder inside the
+                        # zip/rar file, therefore \ should be replaced by / in order to guess it properly
+                        base_name = os.path.splitext(name)[0].replace('\\', '/')
+                        guess = guess_file_info(base_name + '.mkv', type=title.get('type'))
+                        if not self.matches(params, guess):
+                            continue
+
+                        subtitle = LegendasTvSubtitle(language, page_link, subtitle_id, name, guess=guess,
+                                                      imdb_id=title.get('imdb_id'), type=title.get('type'),
+                                                      season=title.get('season'), year=title.get('year'),
+                                                      no_downloads=no_downloads, rating=rating, featured=featured,
+                                                      multiple_episodes=multiple_episodes, timestamp=timestamp)
```

Comment:
Extracting all can lead to memory issues since we're downloading all candidates (15-20 files) and each file can have subtitles for several releases or several episodes. Just a few KB can grow to tens of MB easily. Maybe it's better to keep the binary data and the extraction will only happen if that subtitle was the chosen one.